src/backend/distributed/utils/node_metadata.c

```diff
@@ -478,10 +529,17 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 		deletedNodeId = workerNode->nodeId;
 	}
 
-	DeleteNodeRow(nodeName, nodePort);
-
 	DeleteAllReferenceTablePlacementsFromNode(nodeName, nodePort);
 
+	hasAnyShardPlacements = NodeHasShardPlacements(nodeName, nodePort,
+												   onlyConsiderActivePlacements);
+	if (hasAnyShardPlacements)
+	{
+		ereport(ERROR, (errmsg("you cannot remove a node which has shard placements")));
```

Comment:
@marcocitus I added a call to a newely created `DeleteAllShardPlacementesFromGroup` here which seems to work but surfaces a new error later on:

```
 DROP TABLE cluster_management_test;
+WARNING:  could not find any shard placements for shardId 1220001
+CONTEXT:  SQL statement "SELECT master_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name)"
+PL/pgSQL function citus_drop_trigger() line 21 at PERFORM
```

I think it's probably not safe to also delete any `ShardInterval`s which have been abandoned?